modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/MuleArtifactClassLoader.java

```diff
@@ -110,20 +111,38 @@ public void setResourceReleaserClassLocation(String resourceReleaserClassLocatio
   }
 
   protected ResourceReleaser createResourceReleaserInstance() {
+    return (ResourceReleaser) createCustomInstance(resourceReleaserClassLocation);
+  }
+
+  protected Object createCustomInstance(String classLocation) {
     InputStream classStream = null;
     try {
-      classStream = this.getClass().getResourceAsStream(resourceReleaserClassLocation);
+      classStream = this.getClass().getResourceAsStream(classLocation);
       byte[] classBytes = IOUtils.toByteArray(classStream);
       classStream.close();
       Class clazz = this.defineClass(null, classBytes, 0, classBytes.length);
-      return (ResourceReleaser) clazz.newInstance();
+      return clazz.newInstance();
     } catch (Exception e) {
-      throw new RuntimeException("Can not create resource releaser instance from resource: " + resourceReleaserClassLocation, e);
+      throw new RuntimeException("Can not create instance from resource: " + classLocation, e);
     } finally {
       closeQuietly(classStream);
     }
   }
 
+  /**
+   * Setup reactor-core error hooks, these are required for plugins that end up executing reactive streams.
+   *
+   * For instance, compatibility plugin which inherits from transformers that call {@code block()}.
+   */
+  public void configureErrorHooks() {
```

Comment:
I thought we only wanted to do it when building plugin class loaders and not every time.